source/common/common/interval_value.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <algorithm>
+
+namespace Envoy {
+
+// Template helper type that represents a closed interval with the given minimum and maximum values.
+template <typename T, T MinValue, T MaxValue> struct Interval {
```

Comment:
Done.

Reply:
Seems like the name is still "Interval"

Reply:
I opted to make the change to the value type since it didn't seem helpful to give the interval type an opinion on whether it is open or closed - that is, the value type is templated over a thing with a min and a max, which sounds like it could apply to both an open or closed interval (though "bounds" might be better for an open interval).